content/en/blog/adaptive-syscalls-selection/index.md

```diff
@@ -0,0 +1,212 @@
+
+- `processing`/`parsing`: Extracting meaningful information from the events captured by Falco and converting them into a structured format. We extract data fields in the kernel and perform the necessary parsing in userspace.
+
+- `filtering`: Refers to stopping the processing / parsing or ignoring events. No modifications are made to the kernel during this process.
+
+- `rules matching`: Refers to evaluating an event in userspace against the Abstract Syntax Tree (AST) generated from a Falco rule.
+
+## Background
+
+Before the 0.35.0 release, Falco would monitor a predefined set of commonly used syscalls in its kernel driver for threat detection, regardless of the specific rules being applied. However, this approach had limitations as it would monitor a large number of syscalls. In certain user configurations, Falco would needlessly monitor syscalls not relevant to the loaded rules, consuming system resources without effectively contributing to the intended purpose of threat detection.
+
+You may ask yourself why Falco has been monitoring a predetermined set of commonly used syscalls until now. Falco relies on a set of syscalls to establish and maintain its state in userspace. For example, when a new process is spawned or a network connection is created, multiple syscalls are involved. Additionally, Falco maintains a process cache table in userspace, which requires tracking certain syscalls to ensure the accuracy and currency of the cache table. The process table is crucial for retrieving real-time process tree lineages and other functions. 
+
+Initially, tracing a predefined set of syscalls provided a solid foundation for Falco's functionality. However, with the growing computational workload on servers and systems, it became necessary to adopt a new and more efficient approach to optimize performance.
+
+## New Configuration Options
+
+Falco continues to operate as usual, adhering to its standard behavior. Starting with Falco version 0.35.0, the changes primarily impact the handling of syscall events and the selection of specific syscalls to be monitored and analyzed during live captures.
+
+The behavior of Falco will vary depending on the event source, with changes only affecting live syscall events:
+
+- **non syscall event source**: Falco behaves as usual, where for example container events are retrieved and processed from the container runtime, just like before.
+- **syscall event source**:
+  1. Falco determines a base set of syscalls to monitor. This can be either the default minimum set known as the "sinsp state set" (automatically determined by the underlying libsinsp library) to maintain state consistency and stability, or a customized set of syscalls defined by the user via the new `base_syscalls` configuration.
```

Comment:
```suggestion
  1. Falco determines a base set of syscalls to monitor. This can be either the default minimum set known as the "sinsp state set" (automatically determined by the underlying libsinsp library to maintain state consistency and stability), or a customized set of syscalls defined by the user via the new `base_syscalls` configuration (discussed later).
```